### PR TITLE
Add rfc for k8s multi cluster deployment

### PR DESCRIPTION
**Pull request by ffjlabo**
**What this PR does / why we need it**:

as title

**Which issue(s) this PR fixes**:

Part of #5006

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: docs/rfcs/0014-multi-cluster-deployment-for-k8s.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+# Summary
+
+This RFC proposes a new feature for k8s app to deploy resources into multi-cluster.
+
+# Motivation
+
+# Usecase
+- case 1. When applying the same manifest to multiple clusters for redundant configuration
+- case 2. When applying manifest with some patches applied to multiple clusters for redundant configuration
+- case 3. Blue/Green Deployment acress the clusters
```
Comment (t-kikuc) [inline docs/rfcs/0014-multi-cluster-deployment-for-k8s.md:13]: [nits] 
```suggestion
- case 3. Blue/Green Deployment across clusters
```

File: docs/rfcs/0014-multi-cluster-deployment-for-k8s.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+We propose the feature to apply the manifests to multiple-clusters in one application.
+
+![image](assets/0014-pipeline-image.png)
+
+## How it works
+
+### Register Application with multiple platform providers
+
+On the application register part, we can choose multiple platform providers.
+At first, user add the first platform provider.
```
Comment (t-kikuc) [inline docs/rfcs/0014-multi-cluster-deployment-for-k8s.md:28]:
```suggestion
At first, a user adds the first platform provider. 
```

File: docs/rfcs/0014-multi-cluster-deployment-for-k8s.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+### Register Application with multiple platform providers
+
+On the application register part, we can choose multiple platform providers.
+At first, user add the first platform provider. 
+If we want to use the feature for deploying multi-cluster, we can set more platform providers. This is optional.
+Only the platform providers specified here can be configured for multi-target.
+
+![image](assets/0014-choose-multiple-providers.png)
+
+Also, we can can the list of platform providers on the piped list page to verify the platform providers.
```
Comment (t-kikuc) [inline docs/rfcs/0014-multi-cluster-deployment-for-k8s.md:34]: [nits] plz correct the verb. 
```suggestion
Also, we can xxx the list of platform providers on the piped list page to verify the platform providers.
```

File: docs/rfcs/0014-multi-cluster-deployment-for-k8s.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+We introduce the feature into the stages where changes are made to resources on the cluster.
+
+- K8S_PRIMARY_ROLLOUT
+- K8S_CANARY_ROLLOUT
+- K8S_CANARY_CLEAN
+- K8S_BASELINE_ROLLOUT
+- K8S_BASELINE_CLEAN
+- K8S_TRAFFIC_ROUTING
+
+### How to check the stage progress of each platform providers in the deployment
```
Comment (t-kikuc) [inline docs/rfcs/0014-multi-cluster-deployment-for-k8s.md:179]:
```suggestion
### How to check the stage progress of each platform provider in the deployment
```

File: docs/rfcs/0014-multi-cluster-deployment-for-k8s.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+```
+
+- When deploying
+    - Sync all clusters corresponding to prd when triggering the root app.
+    - If you want to sync clusters partially, sync them as the each application.
+
+- When rolling back
+    - Roll back in the all previous state.
+
+- Rollback時
```
Comment (t-kikuc) [inline docs/rfcs/0014-multi-cluster-deployment-for-k8s.md:320]: these lines should be removed??
```suggestion

```

**Review by t-kikuc — COMMENTED** [review]
What a great rfc 💯 

Some nits.

**Comment by ffjlabo** [comment]
@t-kikuc Thank you for the verbal comment! I fixed them 🙏

**Review by t-kikuc — APPROVED** [review]

**Review by khanhtc1202 — APPROVED** [review]

**Pull request merged**